server/principal.go

```diff
@@ -96,6 +109,16 @@ func (s *Server) registerPrincipalRoutes(g *echo.Group) {
 		if err := jsonapi.UnmarshalPayload(c.Request().Body, principalPatch); err != nil {
 			return echo.NewHTTPError(http.StatusBadRequest, "Malformed patch principal request").SetInternal(err)
 		}
+
+		if principalPatch.Type == api.ServiceAccount && principalPatch.Password != nil {
```

Comment:
This may not be right as the frontend won't be able to specify the password. We may need a different field in the request.

Reply:
Frontend can use any random string just like the principal creation. But sure it's better to have another field to make it more clear.